sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py

```diff
@@ -566,11 +565,10 @@ def create_container(
 
     @distributed_trace
     def delete_container(
-            self, container,  # type: Union[ContainerProperties, str]
-            lease=None,  # type: Optional[Union[BlobLeaseClient, str]]
-            **kwargs
-        ):
-        # type: (...) -> None
+            self, container: Union["ContainerProperties", str],
```

Comment:
Let's open a GitHub issue to consider deprecating `ContainerProperties` as well as any other Properties input we have across all packages. I don't think any have a glaring issue like `BlobProperties` does but may be good to be consistent.